src/RL/nn_structures/nn_structures.jl

```diff
@@ -9,18 +10,23 @@ abstract type NNStructure end
 
 Make NNStructure able to work with batches.
 """
-function (nn::NNStructure)(x::AbstractArray{Float32,3})
-    batch_size = size(x)[end]
+(nn::NNStructure)(ts::AbstractTrajectoryState) = throw(ErrorException("missing function (::$(typeof(nn)))(::$(typeof(ts)))."))
+function (nn::NNStructure)(x::AbstractVector{<:TabularTrajectoryState})
+    batch_size = size(x, 3)
     qval = [nn(x[:, :, i]) for i in 1:batch_size]
     hcat(qval...)
```

Comment:
I agree, it's never used in our code and it's unefficient. But it also is a very generic way to define the forward function for a batch of states, and I think it would work in almost all cases when no more specific function is defined.